### ISSUE TITLE
BUG-FIX: wrong comparison of float type numbers in tests 

### ISSUE THREAD

**Issue by sem-geologist** [hyperspy/hyperspy#957]
There are "random" travis failures which are caused by not right comparison method of  python floats (which are double?). To better understand the issue read http://www.cygnus-software.com/papers/comparingfloats/comparingfloats.htm .

So generally the floating points should not be compared like this:
```0.5 == 0.5``` as depending from compiler, architecture, or even os scheduler, that can return False.
I noticed that most of random failing tests use ```assert_equal``` (which under hood does '==') from ```nose.tools```, it should be changed to ```assert_almost_equal(result, expected_val, places=sane_number_of_digits)```. In some cases where ```assert_equal``` is used on lists or dicts containing floating points, this will get complicated and will need introducing loops to iterate, or the tests should be redesigned. There are some cases in tests where assert_true(float_val) is used, which is also inappropriate.

TODO:
* [x] identify as much as possible dangerous tests
* [x] fix it
* [x] add caveat to dev documentation about float and tests to prevent it in future #958

Bellow I am going to list all affected tests by wrong test methods on float type numbers.

**Comment by sem-geologist** [comment]
The test list:

test (without 'hyperspy.tests.') part | fix | state
-----|-----|-------
signal.test_eds_sem.test_take_off_angle | ```assert_equal``` to ```assert_almost_equal``` | addressed and fixed in #949
model.test_fancy_indexing. TestModelIndexing. test_model_signal_indexer_slice | that is complicated, but the probable cause is the ```assert_true``` on ```float``` number :open_mouth: | addressed at #914. Not fixed.

**Comment by sem-geologist** [comment]
I find out that:
with ```sys.float_info.dig``` it is possible to get python float type precision on the given system.
I.e. on my debian 64bit machine it is 15. probably I should change offending assert_equal parts to:
```python
 assert_almost_equal(measured, expected,
                     places=sys.float_info.dig-(expected.__int__() // 10 + 1 ))
```

by using such approach instead of fixed values it will be more flexible and should stop randomly failing on windows32 builds.

**Comment by francisco-dlp** [comment]
Great bug report!

**Comment by tjof2** [comment]
I've not read `assert_almost_equal` in detail, but I presume it implements the comparison checks appropriately!

**Comment by dnjohnstone** [comment]
what remains to be done for this?

**Comment by sem-geologist** [comment]
the #949, which I am stuck with...

**Comment by francisco-dlp** [comment]
Upon inspection of the test that is randomly failing, I don't think that it is related to float comparisons... Can  we close this then?

**Comment by sem-geologist** [comment]
It is! look at table: 1a) this does not fail at linux travis, but fails at OSX. The #949 is fixing this. Issue can be closed only then #949 will get sorted out.

**Comment by francisco-dlp** [comment]
@sem-geologist and @bm424, we are about to release HyperSpy 0.8.5 and you'll get credited as an author for your contributions to HyperSpy in our zenodo DOI. However, Zenodo takes the author info from your github profile. If you care about being credited to your real name, could you update your github profile with your real name and affiliation?

Thanks!

**Comment by sem-geologist** [comment]
@francisco-dlp , I should not be added to contribution until my contributions will be merged (bruker bcf reader and cythonization/ travis osx builds). However I see this coming release as very good oportunity to give me the detsils how to test git release with appveyor :), so I could finish my opened contributions.

**Comment by francisco-dlp** [comment]
@sem-geologist, your contributions will be released in HyperSpy 1.0, not in the 0.8.5. However, zenodo takes the authors from the master branch, so you'll be credited in the 0.8.5 release despite the above.

**Comment by sem-geologist** [comment]
as #949 got merged, I am closing this
